### PR TITLE
Bugfix/172213552 explorer races

### PR DESCRIPTION
**Pull request by j16r**
Left explorer CI on running for a couple of hours and had this one failure.

### CODE REVIEW

File: explorer/jest.setup.ts [hunk 1]
```diff
@@ -10,5 +10,7 @@ beforeAll(async () => {
 })
 afterEach(() => clearDb())
 afterAll(async () => {
-  await globalDBConnection.close()
+  if (globalDBConnection) {
```
Comment (HenryNguyen5) [inline explorer/jest.setup.ts:13]: This should always exist no?

File: explorer/src/__tests__/sessions.test.ts [hunk 2]
```diff
@@ -19,7 +19,18 @@ describe('sessions', () => {
...
-      expect(foundSession.finishedAt).toBeDefined()
+      expect(foundSession.finishedAt).toEqual(expect.any(Date))
+    })
+
+    it('closes a session', async () => {
+      const [chainlinkNode, secret] = await createChainlinkNode(
+        'valid-chainlink-node',
+      )
+      const session = await authenticate(chainlinkNode.accessKey, secret)
+      expect(session).toBeDefined()
```
Comment (HenryNguyen5) [inline explorer/src/__tests__/sessions.test.ts:30]:
```suggestion
      expect(session).toEqual(expect.any(Session))
```

File: explorer/src/__tests__/sessions.test.ts [hunk 3]
```diff
@@ -35,4 +46,22 @@ describe('sessions', () => {
...
+      expect(session.chainlinkNodeId).toEqual(chainlinkNode.id)
+      expect(session.finishedAt).toBeNull()
+
+      await closeSession(session)
+
+      const foundSession = await getRepository(Session).findOne()
```
Comment (HenryNguyen5) [inline explorer/src/__tests__/sessions.test.ts:62]:
```suggestion
      const foundSession = await getRepository(Session).findOne(session.id)
```

File: explorer/src/__tests__/sessions.test.ts [hunk 4]
```diff
@@ -19,7 +19,18 @@ describe('sessions', () => {
...
+
+    it('closes a session', async () => {
+      const [chainlinkNode, secret] = await createChainlinkNode(
+        'valid-chainlink-node',
+      )
+      const session = await authenticate(chainlinkNode.accessKey, secret)
+      expect(session).toBeDefined()
+      expect(session.chainlinkNodeId).toEqual(chainlinkNode.id)
+
+      closeSession(session)
```
Comment (HenryNguyen5) [inline explorer/src/__tests__/sessions.test.ts:33]:
```suggestion
      await closeSession(session)
```

File: explorer/jest.setup.ts (hunk 1 above)
Comment (HenryNguyen5) [reply explorer/jest.setup.ts:13]: Maybe we should remove `globalDBConnection` altogether and replace it with `getConnection().close()`. It'll throw an error that way too that its not connected

Comment (j16r) [reply explorer/jest.setup.ts:13]: I don't think `getConnection().close()` actually closes the pool, just that connection.

Comment (HenryNguyen5) [reply explorer/jest.setup.ts:13]: The connection is the pool 
image

File: explorer/jest.setup.ts [hunk 5]
```diff
@@ -1,16 +1,16 @@
...
 beforeAll(async () => {
-  globalDBConnection = await openDbConnection()
+  await openDbConnection()
 })
 afterEach(() => clearDb())
 afterAll(async () => {
-  if (globalDBConnection) {
-    await globalDBConnection.close()
+  const db = getConnection()
+  if (db) {
```
Comment (HenryNguyen5) [inline explorer/jest.setup.ts:13]: The connection will always exist, or throw an exception.

Comment (j16r) [reply explorer/jest.setup.ts]: !!!!

File: explorer/jest.setup.ts [hunk 6]
```diff
@@ -9,8 +9,9 @@ beforeAll(async () => {
 })
 afterEach(() => clearDb())
 afterAll(async () => {
-  const db = getConnection()
-  if (db) {
-    await db.close()
+  try {
+    await getConnection().close()
+  } catch (e) {
```
Comment (HenryNguyen5) [inline explorer/jest.setup.ts:14]:
```suggestion
  } catch {
   return
```

File: explorer/src/entity/Session.ts [hunk 7]
```diff
@@ -33,33 +30,30 @@ export class Session {
 
 export async function createSession(
   node: ChainlinkNode,
-  manager?: EntityManager,
+  manager = getManager(),
```
Comment (rupurt) [inline explorer/src/entity/Session.ts:33]: 👍

**Pull request merged**